nodestream/pipeline/extractors/stores/aws/dynamodb_extractor.py

```diff
@@ -0,0 +1,199 @@
+    """
+    def __init__(self, object: Dict[str, Any]):
+        self.attribute_type = list(object.keys())[0]
+        self.value = list(object.values())[0]
+
+    def items(self) -> tuple[str, Any]:
+        return self.attribute_type, self.value
+    
+    def as_dynamo_representation(self):
+        if isinstance(self.value, dict):
+            return {self.attribute_type: {key: value.as_dynamo_representation() for key, value in self.value.items()}}
+        elif isinstance(self.value, list):
+            return {self.attribute_type: [value.as_dynamo_representation() for value in self.value ]}
+        return {self.attribute_type: self.value}
+
+def leave_untouched(value):
+    return value
+
+def convert_scalar(value, type_handler):
+    return  None if type_handler is None else type_handler(value)
+
+def convert_array(array: List[Any], type_handler):
+    return [type_handler(value) for value in array]
+
+def convert_map(mapping: Dict[str, Any], type_handler):
+    return {key: type_handler(value) for key, value in mapping.items()}
+
+def convert_number(value: str):
+    if '.' in value:
+        return float(value)
+    return int(value)
+
+DEFAULT_HANDLER = (convert_scalar, leave_untouched)
+
+def convert_value(dynamo_object: Dict[str, Any]):
+    attribute_type, value = DynamoObject(dynamo_object).items()
+    result_handler, type_handler = ATTRIBUTE_TYPE_DESCRIPTOR_HANDLERS.get(attribute_type, DEFAULT_HANDLER)
+    return result_handler(value, type_handler)
+
+
+ATTRIBUTE_TYPE_DESCRIPTOR_HANDLERS = {
+    "S": (convert_scalar, str),
+    "N": (convert_scalar, convert_number),
+    "B": (convert_scalar, bytes),
+    "NULL": (convert_scalar, None),
+    "BOOL": (convert_scalar, bool),
+    "SS": (convert_array, str),
+    "NS": (convert_array, convert_number),
+    "BS": (convert_array, bytes),
+    "M": (convert_map, convert_value),
+    "L": (convert_array, convert_value)
+}
+
+class DynamoRowConverter:
+    def __init__(self, item: Dict[str, Any]) -> None:
+        self.item = item
+
+    # { key: {"S": "string"}, secondary_key: { "M": {Stuff} }, tertiary_key: { "L": [Stuff] } }
+    def convert_row(self):
+        return {attribute_name: convert_value(attribute_value) for attribute_name, attribute_value in self.item.items()}
+
+
+def verify_list_consistency(value_list: List[Any]):
+    type_set = set([type(value) for value in value_list])
+    if len(type_set > 1):
+        return "L"
+    return REVERSE_ORDER_DESCRIPTOR_HANDLERS[type_set[0]] + "S"
+
+def boolean_to_dynamo_format_converter(value: None | bool):
+    return True if value is None else False
+
+def convert_value_to_dynamo_format(value: Any):
+    value_type, handler = REVERSE_ORDER_DESCRIPTOR_HANDLERS[type(value)]
+    if (value_type == "LIST"):
+        value_type = verify_list_consistency(value)
+    converted_value = handler(value)
+    return DynamoObject({value_type: converted_value})
+
+def recursive_dynamo_format_converter(value: Dict[str, Any] | List[Any]):
```

Comment:
These appear to only be used in the map below.... Why not just make this two functions? One that handles lists and the other that handles maps. Having the if check here makes things less explicit.